### PR TITLE
configs/nxp: use the common partition method to configure partitions.

### PR DESCRIPTION
**Pull request by jeongarmy**
use the common partition method instead of NXP specific method.

### CODE REVIEW

File: build/configs/imxrt1050-evk/loadable_elf_apps/defconfig [hunk 1]
```diff
@@ -296,7 +293,16 @@ CONFIG_IMXRT_HYPERFLASH=y
...
-# CONFIG_ARCH_USE_FLASH is not set
+CONFIG_ARCH_USE_FLASH=y
+
+#
+# Board-Partition Options
+#
+CONFIG_FLASH_PARTITION=y
+CONFIG_FLASH_MINOR=0
+CONFIG_FLASH_PART_SIZE="256,256,256,256,4096,"
+CONFIG_FLASH_PART_TYPE="none,none,ftl,ftl,smartfs,"
```
Comment (yashwanth686007) [inline build/configs/imxrt1050-evk/loadable_elf_apps/defconfig]: Smartfs Partition is created but is not enabled in this defconfig. Please enable smartfs.

**Comment by seinfra** [comment]
Target : [78608d439c53921c2993c4e400385947dbd33477] - Code Rule Check Result:
os/board/imxrt1020-evk/src/imxrt_bringup.c:227: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/imxrt1020-evk/src/imxrt_bringup.c:228: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/imxrt1020-evk/src/imxrt_bringup.c:231: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Target : [78608d439c53921c2993c4e400385947dbd33477] - Code Rule Check Result:
os/board/imxrt1020-evk/src/imxrt_bringup.c:227: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/imxrt1020-evk/src/imxrt_bringup.c:228: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/imxrt1020-evk/src/imxrt_bringup.c:231: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Target : [78608d439c53921c2993c4e400385947dbd33477] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [78608d439c53921c2993c4e400385947dbd33477] - Code Rule Check (C++) OK.

File: build/configs/imxrt1050-evk/loadable_elf_apps/defconfig (hunk 1 above)
Comment (jeongarmy) [reply build/configs/imxrt1050-evk/loadable_elf_apps/defconfig]: It is not related to this commit. will modify later.

**Comment by sunghan-chang** [comment]
@jeongarmy Don't you need to include new header, common.h?

**Comment by seinfra** [comment]
Target : [ac23e659cd54921850330983c361c406bb2c095b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ac23e659cd54921850330983c361c406bb2c095b] - Code Rule Check Result:
os/board/imxrt1020-evk/src/imxrt_bringup.c:228: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/imxrt1020-evk/src/imxrt_bringup.c:229: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/imxrt1020-evk/src/imxrt_bringup.c:232: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

File: os/board/imxrt1020-evk/Kconfig [hunk 2]
```diff
@@ -14,43 +14,12 @@ config IMXRT_NORFLASH
 config IMXRT_FLASH_PART
```
Comment (sunghan-chang) [inline os/board/imxrt1020-evk/Kconfig]: This is duplicated with FLASH_PARTITION. Let's remove this together.

**Comment by seinfra** [comment]
Target : [b69bbbe7ceb394b0e25ece836c2541c900b965c8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b69bbbe7ceb394b0e25ece836c2541c900b965c8] - Code Rule Check Result:
os/board/imxrt1020-evk/src/imxrt_bringup.c:230: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/imxrt1020-evk/src/imxrt_bringup.c:231: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/imxrt1020-evk/src/imxrt_bringup.c:234: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Pull request merged**